superset/common/query_context_processor.py

```diff
@@ -571,11 +574,22 @@ def get_data(self, df: pd.DataFrame) -> str | list[dict[str, Any]]:
                     df, index=include_index, **config["CSV_EXPORT"]
                 )
             elif self._query_context.result_format == ChartDataResultFormat.XLSX:
-                result = excel.df_to_excel(df, **config["EXCEL_EXPORT"])
+                ndf = self.copy_df_for_excel_export(df, coltypes)
```

Comment:
@squalou Should this change be actually implemented in `superset.utils.excel.df_to_excel` to ensure consistent behavior? You could move your tests to `tests/unit_tests/utils/excel_tests`. We already have one there called `test_timezone_conversion` which is a similar type of conversion.

Reply:
Makes a lot of sense, sure.

What would be better,
add "coltypes" as optional parameter in excel.df_to_excel(...) method ?
Or move the method as-is in superset.utils.excel. package.

First option seems better but I'm not at ease with changing method signatures there without an advice, I really have no global overview of what may call what and consequences.